Doc 2451 ci 287xx (#377)
harness/developer-hub: release-notes/continuous-integration.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## January 17, 2023, version 78201

### What's new

* This release includes two new CI steps for integrating your Harness CI pipelines with other GitHub Actions and Bitrise. These actions are available in Harness Cloud build infrastructures only. 
   - An Actions step that you can use to run GitHub Actions.
   - A Bitrise step that enables you to run Bitrise steps and workflows. (CI-6479)

> * Harness CI now supports remote debugging of remote builds in Harness Cloud, Kubernetes, and VMs in AWS, Azure, and other cloud platforms. If a build fails or times out, you can re-run it in debug mode. This option is available in the **Builds**, **Execution**, and **Execution History** pages of the Harness UI. (CI-6350) 

Reviewer: If a build fails at run steps. we should remove timeout.

> * This release includes two new CI steps for integrating your Harness CI pipelines with other GitHub Actions and Bitrise. These actions are available in Harness Cloud build infrastructures only. 
>    - An Actions step that you can use to run GitHub Actions.
>    - A Bitrise step that enables you to run Bitrise steps and workflows. (CI-6479)
> * Harness CI now supports remote debugging of remote builds in Harness Cloud, Kubernetes, and VMs in AWS, Azure, and other cloud platforms. If a build fails at a Run step, you can re-run the build in debug mode. This option is available in the **Builds**, **Execution**, and **Execution History** pages of the Harness UI. (CI-6350) 

Reviewer 2: ```suggestion
* This release includes two new CI steps for integrating your Harness CI pipelines with GitHub Actions and Bitrise. The following steps are available in Harness Cloud build infrastructures only. 
   - An Actions step to run GitHub Actions.
   - A Bitrise step to run Bitrise steps and workflows. (CI-6479)
* Harness CI now supports remote debugging of remote builds in Harness Cloud, Kubernetes, and VMs in AWS, Azure, and other cloud platforms. If a build fails at a Run step, you can rerun the build in debug mode. This option is available in the **Builds**, **Execution**, and **Execution History** pages of the Harness UI. (CI-6350) 
```

Reviewer 2: - not familiar with the term Harness Cloud. Is this what we call CI builds done in the build farm Harness hosts?
- I looked at my CI module and I didn't see **Builds**, **Execution**, and **Execution History** pages. Not sure where you mean.

   
  ![](./static/ci-rerun-build-in-debug-mode.png) 

> * You can now specify hostnames in Kubernetes build infrastructures, which enables your pipelines to communicate with external services using hostnames instead of IPs. The following YAML snippet shows how to set up aliases for your cluster. (CI-5996, ZD-36578)

Reviewer 2: ```suggestion
* You can now specify hostnames instead of IPs in Kubernetes build infrastructures. This enables your pipelines to communicate with external services using hostnames. The following Harness YAML snippet shows how to set up aliases for your cluster in the CI stage **Infrastructure** section. (CI-5996, ZD-36578)
```
